### PR TITLE
ddtrace/tracer: obfuscate stats

### PR DESCRIPTION
**Pull request by gbbr**
This change introduces obfuscation of stats aggregation key resources by
using the same code as used in the trace-agent.

This feature continues to be off by default, enablable using the `discovery` feature flag.

### CODE REVIEW

**Review by piochelepiotr — APPROVED** [review]

**Review by piochelepiotr — APPROVED** [review]

File: ddtrace/tracer/option.go [hunk 1]
```diff
@@ -359,23 +368,29 @@ func (c *config) loadAgentFeatures() {
...
 	for _, endpoint := range info.Endpoints {
 		switch endpoint {
 		case "/v0.6/stats":
-			if c.HasFeature("discovery") {
-				// client-stats computation is off by default
-				c.features.Stats = true
+			// disable_stats feature flag is here temporarily to allow reverting
+			// in case of any issues.
+			if !c.HasFeature("disable_stats") {
+				c.agent.Stats = true
```
Comment (pkalmakis) [inline ddtrace/tracer/option.go:386]: We need to prove this feature out with real production customers before turning it on by default for everyone.

**Review by pkalmakis — CHANGES_REQUESTED** [review]
Let's please hold off on enabling client-computed stats by default. We currently have 0 production customers using this feature. We should get a few to validate that the go tracer is working properly before enabling this by default for 100% of customers who upgrade.

File: ddtrace/tracer/span.go [hunk 2]
```diff
@@ -398,6 +399,31 @@ func newAggregableSpan(s *span, cfg *config) *aggregableSpan {
...
+			return textNonParsable
+		}
+		return oq.Query
+	case "redis":
+		return o.QuantizeRedisString(resource)
+	default:
+		return resource
+	}
+}
+
```
Comment (knusbaum) [inline ddtrace/tracer/span.go:426]: 🥳  🎉

File: go.mod [hunk 3]
```diff
@@ -3,6 +3,7 @@ module gopkg.in/DataDog/dd-trace-go.v1
 go 1.12
 
 require (
+	github.com/DataDog/datadog-agent/pkg/obfuscate v0.0.0-20211129110424-6491aa3bf583
```
Comment (knusbaum) [inline go.mod:6]: Any chance we can get a proper mod version here?

**Review by knusbaum — COMMENTED** [review]
Looks great!

One question about the obfuscation module.
Let's figure out whether this is enabled/disabled by default and then get it in.

File: go.mod (hunk 3 above)
Comment (gbbr) [reply go.mod:6]: Not yet. We'll have to use the pseudo-version format until the Datadog Agent is released, and an official tag is assigned to this module. It should be fine. Do you see any issues with this approach, until that happens?

**Comment by gbbr** [comment]
@pkalmakis that's done in the latest commit. PTAL

File: ddtrace/tracer/option.go (hunk 1 above)
Comment (gbbr) [reply ddtrace/tracer/option.go]: Done in https://github.com/DataDog/dd-trace-go/pull/1069/commits/719c849101c8a09066fc782ff544a67a52485d33. Thanks for jumping in.

**Review by piochelepiotr — DISMISSED** [review]

**Review by piochelepiotr — APPROVED** [review]

**Review by pkalmakis — APPROVED** [review]

**Pull request merged**